extensions/wikia/PageShare/PageShareHooks.class.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+class PageShareHooks {
+
+	/**
+	 * Add global JS variable wgAcceptLangList
+	 *
+	 * @param array $vars global variables list
+	 * @return boolean return true
+	 */
+	public static function onMakeGlobalVariablesScript(Array &$vars) {
+		global $wgRequest;
+
+		$wgAcceptLang = $wgRequest->getAcceptLang();
+		$vars['wgAcceptLangList'] = is_array( $wgAcceptLang ) ? array_keys( $wgAcceptLang ) : null;
```

Comment:
are you sure it's not gonna be cached?
From what I know we are cashing base on URL.
Imagine situation:
User with header ZH goes to http://muppet.wikia.com -> page gets cached
User with header PL goes to http://muppet.wikia.com -> wgAcceptLangList is cached and is set to ZH